packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -239,7 +240,29 @@ class TestCommand extends FastFlutterCommand {
     if (build != 0) {
       throwToolExit('Error: Failed to build asset bundle');
     }
-    await writeBundle(fs.directory(fs.path.join('build', 'unit_test_assets')), assetBundle.entries);
+    if (_needRebuild(assetBundle.entries)) {
```

Comment:
In theory, the expensive part should be```
 await writeBundle(fs.directory(fs.path.join('build', 'unit_test_assets')),
          assetBundle.entries);
```
I check the code assetBundle.build() is just bunch of string operation, it only loads the pubspec files which we cannot avoid anyway.

Reply:
Lets double check the performance of a single test on gallery, since that has a reasonable number of assets.

The benchmark runs on the hello_world test, so we should gather numbers for that too.

Reply:
would you suggest just time several run with and without the patch?

Reply:
sounds like a good approach

Reply:
I am seeing a little bit improvement
gallery smoke test   113.4 -> 107
hello world 2.98 -> 2.86

Reply:
What are the numbers before the original patch?